plugins/modules/foreman_installation_medium.py

```diff
@@ -116,7 +116,8 @@ def main():
             module.fail_json(msg="'state: present_with_defaults' and 'name: *' cannot be used together")
         if module.desired_absent:
             if list(entity_dict.keys()) != ['name']:
-                module.fail_json(msg='When deleting all installation media, there is no need to specify further parameters %s ' % entity_dict.keys())
+                entity_dict.pop('name', None)
+                module.fail_json(msg='When deleting all installation media, there is no need to specify further parameters: %s ' % entity_dict.keys())
```

Comment:
Removing an entity doesn't feel very clean. Would the following work and provide useful feedback?
```suggestion
                module.fail_json(msg='When deleting all installation media, only name may be passed as a parameter', passed=entity_dict.keys())
```

Reply:
Do you have documentation for this. I cannot find any about how `passed` is used here.

Reply:
It is entirely untested but a passing module essentially returns a dict where `msg` is a bit of a special key. When using the YAML output, you can see all keys.

Reply:
It's not that common in the Ansible codebase, but it does show up:
https://github.com/ansible/ansible/blob/c2469648e4235deb10e3c314bb668e412f2d6ffb/lib/ansible/modules/cloud/vmware/vsphere_file.py#L203-L206

Reply:
I liked the concept of only showing the offending keys. And since we exit in the next line anyway, it seemed appropriate to do it the easy way. This list is already lacking a lot of the actually passed parameters (username, password, server_url, ...).

Reply:
I don't feel very strongly about this. Exactly for the reasons you mention I debated it. If this becomes a pattern then we can add a helper:
```python
def only_allow_params(params, allowed):
    params = set(params)
    allowed = set(allowed)
    if params > allowed:
        message = 'Only %s can be passed; %s passed' % (', '.join(allowed), ', '.join(params))
        raise ValueError(message)
```

However, let's merge this for now.

Reply:
I have seen two other places, that just did not report the superfluous parameters.
At some point, we should look how much of this whole affects_multiple can be dryed up.